tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -405,8 +405,292 @@ bool hasOuterSequentialMember(
   }
   return false;
 }
+
+// Intersect the union set with all the mapping
+// filters params in the given schedule tree
+isl::union_set intersectMappingFilterParams(
```

Comment:
I don't quite understand what this function returns if there are multiple _divergent_ thread mappings in different children, for example one branch is mapped to (32,4) threads and another to (7,1) threads.  Some reasoning in the comment would be helpful.

Reply:
This function add the constraints contained in the mapping filters to the active domain points in the mapping filter node.
If we have a domain { S0[i0] : 0 <= i0 < 42;   S1[i1] : 0 <= i1 < 27 },
followed by a sequence splitting S0 and S1 with filters, and a mapping filter in each branch mapping respectively t0 to i0 on 4 threads and t0 to i1 on 3 threads.
The result would be { S0[i0] : 0 <= i0 < 42 and t0 = i0 mod 4 ;  S1[i1] : 0 <= i1 < 27 and t0 = i1 mod 3 }